core/shared/src/main/scala/zio/prelude/ZValidation.scala

```diff
@@ -218,6 +218,16 @@ sealed trait ZValidation[+W, +E, +A] { self =>
       ZIO.succeedNow
     )
 
+  final def unsafeGetOrThrow: A = this match {
```

Comment:
I don't think I would expose unsafe variants here. We could instead make this something like `toEitherThrowable` that would return an `Either[Throwable, A]` by just wrapping the failures in a new subtype of `Throwable`.

Reply:
OK, I've dropped it and added `toEitherException`.
But I think this will be an common need for our users to assert that the Validation succeeded and throw in the exceptional case that it didn't.
Do you have any idea how to do it elegantly and without boilerplate on the side of our users?

Reply:
I think the most typical thing for a user to do would be something like:

```scala
myValidation.fold(
  errors => throw new MyErrorType(s"validation of myDataType in myModule failed with errors: $errors.toString")
  value  => ???
}
```

We could potentially clean this up slightly with something like `getOrElse`:

```scala
trait ZValidation[+W, +E, +A] { self =>
  def getOrElse[A1 >: A](f: NonEmptyChunk[E] => A1): A1 =
    self.fold(f, identity)
}
```

Then you could do:

```scala
val a = myValidation.getOrElse { errors =>
 thrown new MyErrorType(s"validation of myDataType in myModule failed with errors: $errors.toString)
}
```

That is more general in that it allows the user to handle the error however they like and also potentially lets us avoid having to define this one particular subtype of `Throwable` that is not going to be what the user wants most of the time. `Either` for example doesn't define some built in way for users to convert arbitrary error types to `Throwable` but leaves that up to the user.

Reply:
I've added `getOrHandleException` to _handle_ this use case :smile_cat: